src/ascent/runtimes/expressions/ascent_expression_filters.cpp

```diff
@@ -1382,35 +1382,43 @@ ScalarGradient::execute()
   bool cycles = units == "cycle";
   int total = execution_points + time + cycles;
 
-  if(total == 0 && !n_window_length_unit.dtype().is_empty()) {
-     ASCENT_ERROR("ScalarGradient: if a ``window_length_unit`` value is provided, it must be set to either: 1). \"index\", 2). \"time\", or 3). \"cycle\"." );
+  if(total == 0 && !n_window_length_unit.dtype().is_empty())
+  {
+     ASCENT_ERROR("ScalarGradient: if a ``window_length_unit`` value is provided,"
+                  <<" it must be set to either: 1). \"index\", 2). \"time\", or 3). \"cycle\"." );
   }
+
   if((execution_points || cycles) && window_length < 1) {
      ASCENT_ERROR("ScalarGradient: window_length must be at least 1 if the window length unit is \"index\" or \"cycle\"." );
   }
 
   const conduit::Node &history = (*cache)[expr_name];
 
   const int entries = history.number_of_children();
-  if(entries < 2) {
+  if(entries < 2)
+  {
     (*output)["value"] = -std::numeric_limits<double>::infinity();
     (*output)["type"] = "double";
     set_output<conduit::Node>(output);
     return;
   }
 
-  int first_index = -1, current_index = entries - 1;
-  if(execution_points) {
+  int first_index = 0, current_index = entries - 1;
```

Comment:
This is what I changed (other than some formatting).

Reply:
Good catch. first_index needed to be set to -1 for a subsequent expression, but not for this one.